### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by RomanHrytsiuk**
https://RomanHrytsiuk.github.io/layout_product-cards/
https://RomanHrytsiuk.github.io/layout_product-cards/report/html_report/

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,13 +1,53 @@
...
+  </head>
+
+  <body>
+    <div class="card" data-qa="card">
+      <img
+      class="card__image"
+      src="./images/imac.jpeg"
+      alt="APPLE IMac 27 Retina 5K Monoblock">
+
+      <div class="title">
```
Comment (Viktor-Kost) [inline src/index.html:22]: This is redundant div wrapper
```suggestion

```

File: src/index.html [hunk 2]
```diff
@@ -1,13 +1,53 @@
...
+        <h2 class="title__name">
```
Comment (Viktor-Kost) [inline src/index.html:23]: Please, check requirements and apply for the WHOLE document
There should be only 2 BEM blocks in your code card and stars all the others are just card elements

File: src/index.html [hunk 3]
```diff
@@ -1,13 +1,53 @@
...
+      <div class="price">
+        <p class="price__text">Price:</p>
+        <p class="price__dollar">$2,199</p>
+      </div>
+
+      <a
+          class="button"
+          data-qa="hover"
+          href="/">Buy
+      </a>
```
Comment (Viktor-Kost) [inline src/index.html:50]:
```suggestion
      <a
         class="button"
         data-qa="hover"
         href="/">Buy
      </a>
```

File: src/index.html [hunk 4]
```diff
@@ -1,13 +1,53 @@
...
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
+  </head>
+
+  <body>
+    <div class="card" data-qa="card">
+      <img
+      class="card__image"
+      src="./images/imac.jpeg"
+      alt="APPLE IMac 27 Retina 5K Monoblock">
```
Comment (Viktor-Kost) [inline src/index.html:20]:
```suggestion
      <img
        class="card__image"
        src="./images/imac.jpeg"
        alt="APPLE IMac 27 Retina 5K Monoblock"
      >
```

File: src/style.css [hunk 5]
```diff
@@ -0,0 +1,36 @@
+@import "./styles/title.css";
+@import "./styles/stars.css";
+@import "./styles/price.css";
+@import "./styles/cardWithImage.css";
```
Comment (Viktor-Kost) [inline src/style.css:4]: Please, use one case of naming files, naming styles etc [link](https://betterprogramming.pub/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841)

File: src/style.css [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+.button {
+  box-sizing: border-box;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border-radius: 5px;
+  border: 1px solid #00acdc;
+  background-color: #00acdc;
```
Comment (Viktor-Kost) [inline src/style.css:22]: Use variables for colors you use several times

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 7]
```diff
@@ -1,13 +1,51 @@
...
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__review">Review: 5</div>
+      </div>
+
+      <div class="card-price">
+        <p class="card-price__text">Price:</p>
+        <p class="card-price__dollar">$2,199</p>
```
Comment (AlisaVlasova) [inline src/index.html:41]:
```suggestion
      <div class="card__price">
        <p class="card__price-text">Price:</p>
        <p class="card__price-dollar">$2,199</p>
```

File: src/index.html [hunk 8]
```diff
@@ -1,13 +1,51 @@
...
+      </div>
+
+      <a
+        class="button"
```
Comment (AlisaVlasova) [inline src/index.html:45]:
```suggestion
        class="card__button"
```

File: src/style.css [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+html {
+  font-family: Roboto, Helvetica, sans-serif;
+}
+
+body {
+  margin: 0;
+}
+
+.button {
+  box-sizing: border-box;
```
Comment (AlisaVlasova) [inline src/style.css:15]: move it to * selector

File: src/style.css [hunk 10]
```diff
@@ -0,0 +1,35 @@
+@import "./styles/variable.css";
+
+@import "./styles/blocks/card.css";
+@import "./styles/blocks/stars.css";
```
Comment (AlisaVlasova) [inline src/style.css:4]: add these files to git

File: src/style.css [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+  font-weight: 700;
+  color: #fff;
+  margin: 0 16px 16px;
+  text-decoration: none;
+}
+
+.button:hover {
+  background-color: #fff;
+  color: var(--color-blue);
+}
```
Comment (AlisaVlasova) [inline src/style.css:35]: move it to card block

**Review by AlisaVlasova — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 12]
```diff
@@ -1,13 +1,51 @@
...
+
+      <div class="card__price">
+        <p class="card__price-text">Price:</p>
+        <p class="card__price-number">$2,199</p>
+      </div>
+
+      <a
+        class="card__button"
+        data-qa="hover"
+        href="/">Buy
```
Comment (DarkMistyRoom) [inline src/index.html:47]:
```suggestion
        href="#">Buy
```

Comment (DarkMistyRoom) [reply src/index.html:47]: It shouldn't link to 404

File: src/index.html [hunk 13]
```diff
@@ -1,13 +1,51 @@
...
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
+  </head>
+
+  <body>
+    <div class="card" data-qa="card">
+      <img
+        class="card__image"
+        src="./images/imac.jpeg"
+        alt="APPLE IMac 27 Retina 5K Monoblock">
```
Comment (DarkMistyRoom) [inline src/index.html:20]:
```suggestion
      <img
        class="card__image"
        src="./images/imac.jpeg"
        alt="APPLE IMac 27 Retina 5K Monoblock"
      >
```

File: src/index.html (hunk 12 above)
Comment (DarkMistyRoom) [inline src/index.html:47]:
```suggestion
      <a
        class="card__button"
        data-qa="hover"
        href="/"
      >
        Buy
```

File: src/blocks/stars.css [hunk 14]
```diff
@@ -0,0 +1,32 @@
+.stars {
+  display: flex;
+  margin: 0 16px 24px;
```
Comment (DarkMistyRoom) [inline src/blocks/stars.css:3]: Don't set margins to a BEM block

File: src/blocks/stars.css [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+  background-image: url(../images/star-active.svg);
+}
+
+.stars__review {
+  margin-left: 17px;
+  font-size: 10px;
+  line-height: 14px;
+  font-weight: 400;
+  color: var(--color-dark);
+}
```
Comment (DarkMistyRoom) [inline src/blocks/stars.css:32]: It isn't a part of stars block

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]
Your iMac looks a bit flattened
image

You don't have to set so many margins for each element if you set paddings to the card container. Like here
image

File: src/index.html [hunk 16]
```diff
@@ -1,13 +1,54 @@
...
+
+      <div class="stars stars--4">
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__review">Review: 5</div>
+      </div>
+
```
Comment (SanyaBratashchuk) [inline src/index.html:39]:
```suggestion
    <div class="card__reviews">
      <div class="stars stars--4">
        <div class="stars__star"></div>
        <div class="stars__star"></div>
        <div class="stars__star"></div>
        <div class="stars__star"></div>
        <div class="stars__star"></div>
      </div>
      <div class="card__reviews-text">Review: 5</div>
   </div>
```
Make it work like that

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]

File: src/blocks/card.css [hunk 17]
```diff
@@ -0,0 +1,77 @@
+.card {
+  display: flex;
+  flex-direction: column;
+  width: 200px;
+  box-sizing: border-box;
```
Comment (ivanchuk-yurii) [inline src/blocks/card.css:5]: you don't need this here, you already have this in  `*`

File: src/blocks/card.css [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+  line-height: 18px;
+  font-weight: 500;
+  color: var(--color-dark);
+}
+
+.card__code {
+  margin-bottom: 16px;
+  font-size: 10px;
+  line-height: 14px;
+  font-weight: 400;
```
Comment (ivanchuk-yurii) [inline src/blocks/card.css:31]: it's default value, remove, check in other places

File: src/blocks/stars.css [hunk 19]
```diff
@@ -0,0 +1,23 @@
+.stars {
+  display: flex;
+}
+
+.stars__star {
+  border-radius: 0.5px;
+  height: 16px;
+  width: 16px;
+  background-repeat: no-repeat;
+  background-image: url(../images/star.svg);
```
Comment (ivanchuk-yurii) [inline src/blocks/stars.css:10]: it's better to combine this into `background` property

File: src/style.css [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+
+html {
+  font-family: Roboto, Helvetica, sans-serif;
+}
+
+body,
+
+p,
+
+h2 {
```
Comment (ivanchuk-yurii) [inline src/style.css:17]:
```suggestion
body,
p,
h2 {
```

File: src/index.html (hunk 16 above)
Comment (ivanchuk-yurii) [reply src/index.html:39]: not fixed

**Review by ivanchuk-yurii — CHANGES_REQUESTED** [review]

**Review by ivanchuk-yurii — APPROVED** [review]

**Pull request closed without merging**